webview-ui/src/controls/MinecraftStatisticStackedLineChart.tsx

```diff
@@ -0,0 +1,122 @@
+                // set right now
+                const totalValuesPerCatageory: { category: string; value: number }[] = [];
+                newData.forEach(d => {
+                    const currentCategoryIndex = totalValuesPerCatageory.findIndex(c => c.category === d.category);
+                    if (currentCategoryIndex !== -1) {
+                        totalValuesPerCatageory[currentCategoryIndex].value += d.value;
+                    } else {
+                        totalValuesPerCatageory.push({ category: d.category!, value: d.value });
+                    }
+                });
+                totalValuesPerCatageory.sort((a, b) => a.value - b.value);
+
+                newData.sort(
+                    (a, b) =>
+                        totalValuesPerCatageory.findIndex(c => c.category === a.category) -
+                        totalValuesPerCatageory.findIndex(c => c.category === b.category)
+                );
+
+                return newData;
+            });
+        };
+
+        statisticDataProvider.registerWindowListener(window);
+        statisticDataProvider.addSubscriber(eventHandler);
+
+        const latestTime = data.length !== 0 ? data[data.length - 1].time : 0;
+
+        const plot = Plot.plot({
+            color: {
+                legend: true,
+                tickFormat: d => {
+                    const label = catageoryLabels !== undefined ? catageoryLabels[d] : undefined;
+                    if (label !== undefined) {
+                        return label;
+                    } else {
+                        return d;
+                    }
+                },
+            },
+            title: title,
+            x: {
+                label: xLabel,
+                tickFormat: function (d: d3.NumberValue, i: number) {
+                    const tickDifference = latestTime - d.valueOf();
+
+                    if (tickDifference < 20) {
+                        return 'now';
+                    }
+
+                    // Assume 20 ticks per second
+                    return Math.floor(tickDifference / 20) + 's';
+                },
+            },
+            y: { grid: true, label: yLabel, type: 'sqrt' },
+            marks: [
+                Plot.areaY(data, {
+                    x: 'time',
+                    y: 'value',
+                    fill: 'category',
+                    title: 'category',
+                }),
+                Plot.ruleY([0]),
+                targetValue ? Plot.ruleY([targetValue]) : undefined,
+            ],
+        });
+
+        if (containerRef.current !== null) {
+            containerRef.current.append(plot);
+        }
+
+        // Remove old listener
+        return () => {
+            statisticDataProvider.removeSubscriber(eventHandler);
+            statisticDataProvider.unregisterWindowListener(window);
+
+            if (plot !== undefined) {
+                plot.remove();
+            }
+        };
+    });
```

Comment:
Missing dependencies?

Reply:
I mean, the array parameter
```suggestion
    }, [data]);
```